### PR TITLE
Card #7

### PR DESCRIPTION
**Pull request by sonalik-cuelogic**
Card Link: https://trello.com/c/IE6GKagX
Implemented Add Leave

### CODE REVIEW

File: cue_leave_mgmt/urls.py [hunk 1]
```diff
@@ -22,6 +22,7 @@
     url('', include('social.apps.django_app.urls', namespace='social')),
     url('', include('django.contrib.auth.urls', namespace='auth')),
     url(r'^$', 'user_auth.views.index', name='index'),
+    url(r'^leave_mgmt/', include('leave_mgmt.urls', namespace='leave_mgmt')),
     url(r'^employee_details/', include('employee_details.urls', namespace='employee_details')),
```
Comment (onkartibe) [inline cue_leave_mgmt/urls.py]: change this to employee/details (add/edit/list) etc

File: cue_leave_mgmt/urls.py [hunk 2]
```diff
@@ -22,6 +22,7 @@
     url('', include('social.apps.django_app.urls', namespace='social')),
     url('', include('django.contrib.auth.urls', namespace='auth')),
     url(r'^$', 'user_auth.views.index', name='index'),
+    url(r'^leave_mgmt/', include('leave_mgmt.urls', namespace='leave_mgmt')),
```
Comment (onkartibe) [inline cue_leave_mgmt/urls.py]: change this to leave/

File: employee_details/models.py [hunk 3]
```diff
@@ -4,13 +4,14 @@
 
 class User_Profile(models.Model):
     user = models.OneToOneField(User, unique=True)
-    profile_picture = models.CharField(max_length=50)
+    profile_picture = models.CharField(max_length=150)
```
Comment (onkartibe) [inline employee_details/models.py]: what you are planning to store in profile picture?

File: leave_mgmt/templates/leave_mgmt/submit_leaves.html [hunk 4]
```diff
@@ -0,0 +1,13 @@
+{% extends 'base.html' %}
```
Comment (onkartibe) [inline leave_mgmt/templates/leave_mgmt/submit_leaves.html]: this page should not be exist. after success redirect user to view leaves with success message. If error message is there then keep on add leave page.

File: leave_mgmt/urls.py [hunk 5]
```diff
@@ -0,0 +1,7 @@
+from django.conf.urls import include, url
+from leave_mgmt import views
+
+urlpatterns = [
+    url(r'^add/leave/', views.add_leave, name="add_leave"),
```
Comment (onkartibe) [inline leave_mgmt/urls.py]: it should be leave/add

File: leave_mgmt/urls.py [hunk 6]
```diff
@@ -0,0 +1,7 @@
...
+    url(r'^submit/leaves/', views.submit_leaves, name="submit_leaves"),
```
Comment (onkartibe) [inline leave_mgmt/urls.py]: it should be leave/add

File: leave_mgmt/views.py [hunk 7]
```diff
@@ -1,3 +1,73 @@
...
+import datetime
+from datetime import date, timedelta
+from calendar import monthrange, month_name
+from collections import OrderedDict
+from employee_details.models import User_Profile
+from leave_mgmt.models import Employee_Applied_Leaves, Total_Leaves
 
-# Create your views here.
+
+def add_leave(request):
```
Comment (onkartibe) [inline leave_mgmt/views.py]: name of function is not as per standards.

File: leave_mgmt/views.py [hunk 8]
```diff
@@ -1,3 +1,73 @@
...
+    leave_type = Total_Leaves.objects.all()
+    return render(request, 'leave_mgmt/add_leave.html',
+                    {'leave_type':leave_type})
+
+
+def get_from_to_date(leave_plan, data):
```
Comment (onkartibe) [inline leave_mgmt/views.py]: what this function do? it does not justify from name.

**Comment by sonalik-cuelogic** [comment]
All changes made as per review comments.

**Comment by onkartibe** [comment]
merging time being... will do full code review once everything is done.

**Pull request merged**